modules/table/src/main/java/org/apache/ignite/internal/table/distributed/index/IndexBuilder.java

```diff
@@ -43,12 +46,23 @@
 import org.apache.ignite.network.ClusterNode;
 
 /**
- * Class for managing the building of table indexes.
+ * Component that is responsible for building an index for a specific partition.
+ *
+ * <p>Approximate index building algorithm:</p>
+ * <ul>
+ *     <li>If the index has not yet been built ({@link IndexStorage#getNextRowIdToBuild()} {@code != null}) or is not in the process of
+ *     being built, then an asynchronous task is added to build it.</li>
+ *     <li>Index building task generates batches of {@link RowId} (by using {@link IndexStorage#getNextRowIdToBuild()}) and sends this batch
+ *     to the primary replica (only the primary replica is expected to start building the index) so that the corresponding replication group
```

Comment:
So on which node does `IndexBuilder` itself work (if it sends batches to a primary replica)? Is it colocated with the primary replica? How is it guaranteed that only one `IndexBuilder` drives the index build process?